presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotBrokerPageSourceBase.java

```diff
@@ -110,6 +112,30 @@ private static Double parseDouble(String value)
         }
     }
 
+    protected void setValue(Type type, BlockBuilder blockBuilder, JsonNode value)
+    {
+        if (type == null || blockBuilder == null) {
+            return;
+        }
+        if (value == null) {
+            blockBuilder.appendNull();
+            return;
+        }
+        if (type instanceof ArrayType) {
+            checkState(value.isArray());
+
+            BlockBuilder childBuilder = blockBuilder.beginBlockEntry();
+            ArrayNode arrayNode = (ArrayNode) value;
+            for (int i = 0; i < arrayNode.size(); i++) {
+                setValue(((ArrayType) type).getElementType(), childBuilder, asText(arrayNode.get(i)));
```

Comment:
Do we support nested types ? if not, lets somehow explicitly forbid it ? ie arrays inside of arrays.

Reply:
no we don't support nested arrays and it is forbidden by the existing condition within the `setValue` of primitive types,

```
 if (!(type instanceof FixedWidthType) && !(type instanceof VarcharType)) {
            throw new PinotException(PINOT_UNSUPPORTED_COLUMN_TYPE, Optional.empty(), "type '" + type + "' not supported");
        }
```